### PR TITLE
guni - close to dai click triggers form update

### PR DESCRIPTION
**Pull request by piekczyk**
# [Guni form update](https://app.shortcut.com/oazo-apps/story/3208/update-guni-manage-so-that-the-close-needs-to-be-clicked-for-the-order-information-to-show)

<please insert a clubhouse link above>
  
## Changes 👷‍♀️
  <Please add short list of changes>
- user has to click on close to dai button in order to see changes information
  
## How to test 🧪
  <Please explain how to test your changes>
- heroku
    
## Definition of done ✔️

- [ ] Acceptance criteria for each issue met
- [ ] Unit tests written where needed and passing
- [ ] End-to-end tests for happy path
- [ ] Documentation updated <When applicable>
- [ ] Project builds without errors
- [ ] Non-functional requirements met
- [ ] Code reviewed and functionality tested by the reviewer (locally, Heroku, etc.)
- [ ] Feature verified and accepted by Product Owner
- [ ] Project deployed to production environment

### CODE REVIEW

File: features/manageGuniVault/manageGuniVault.ts [hunk 1]
```diff
@@ -109,21 +109,39 @@ function applyGuniCalculations(state: ManageMultiplyVaultState & GuniTxData) {
...
-  const s7 = applyManageVaultCalculations(s6)
-  const s8 = applyGuniCalculations(s7)
-  const s9 = applyManageVaultStageCategorisation(s8 as ManageMultiplyVaultState)
-  const s10 = applyManageVaultConditions(s9)
-  return applyManageVaultSummary(s10)
+  const s3 = applyManageGuniVaultTransition(change as ManageMultiplyVaultChange, s2)
+  const s4 = applyManageVaultTransaction(change as ManageMultiplyVaultChange, s3)
+  const s5 = applyManageVaultEnvironment(change as ManageMultiplyVaultChange, s4)
+  const s6 = applyManageVaultInjectedOverride(change as ManageMultiplyVaultChange, s5)
+  const s7 = applyGuniDataChanges(s6, change as GuniTxDataChange)
```
Comment (adamskrodzki) [inline features/manageGuniVault/manageGuniVault.ts:139]: reversed parameter order, mistake or inconsistency ?

File: features/openMultiplyVault/variants/guni/manage/GuniManageMultiplyVaultForm.tsx [hunk 2]
```diff
@@ -56,7 +56,7 @@ export function GuniManageMultiplyVaultForm(props: ManageMultiplyVaultState) {
           <VaultErrors {...props} />
           <VaultWarnings {...props} />
           {stage === 'manageSuccess' && <VaultChangesWithADelayCard />}
-          <ManageMultiplyVaultButton {...props} />
+          {stage !== 'adjustPosition' && <ManageMultiplyVaultButton {...props} />}
```
Comment (adamskrodzki) [inline features/openMultiplyVault/variants/guni/manage/GuniManageMultiplyVaultForm.tsx:59]: button do not display at all instead of being disabled ?

**Review by adamskrodzki — APPROVED** [review]

File: features/openMultiplyVault/variants/guni/manage/GuniManageMultiplyVaultEditing.tsx [hunk 3]
```diff
@@ -59,15 +59,18 @@ function OtherActionsForm(props: ManageMultiplyVaultState) {
...
+  } = props
 
   return (
     <Grid gap={4}>
       {/* TODO here stage has to be already on otherAction and closeVault value to DAI */}
       {/* originalEditingStage === 'otherActions' && otherAction === 'closeVault' */}
       <OtherActionsForm {...props} />
       {!inputAmountsEmpty && <Divider />}
-      <GuniManageMultiplyVaultChangesInformation {...props} />
+      {!debt.isZero() && <GuniManageMultiplyVaultChangesInformation {...props} />}
```
Comment (adamskrodzki) [inline features/openMultiplyVault/variants/guni/manage/GuniManageMultiplyVaultEditing.tsx:73]: how not showing summary if debt is zero relates to the task ?

**Review by adamskrodzki — CHANGES_REQUESTED** [review]

**Review by adamskrodzki — DISMISSED** [review]

File: features/openMultiplyVault/variants/guni/manage/GuniManageMultiplyVaultForm.tsx (hunk 2 above)
Comment (piekczyk) [reply features/openMultiplyVault/variants/guni/manage/GuniManageMultiplyVaultForm.tsx:59]: The first one - from my point of view displaying button which says "Adjust your position" (even if disabled) doesn't make sense from managing GUNI perspective.

image

File: features/openMultiplyVault/variants/guni/manage/GuniManageMultiplyVaultEditing.tsx (hunk 3 above)
Comment (piekczyk) [reply features/openMultiplyVault/variants/guni/manage/GuniManageMultiplyVaultEditing.tsx:73]: I added this condition due to the fact that showing Changes information in form doesn't make sense when vault is already closed.

File: features/manageGuniVault/manageGuniVault.ts (hunk 1 above)
Comment (piekczyk) [reply features/manageGuniVault/manageGuniVault.ts:139]: I added applyManageGuniVaultTransition after s2 and thats why each state later is renamed.

Comment (adamskrodzki) [reply features/manageGuniVault/manageGuniVault.ts:139]: I ment that all except  applyGuniDataChanges
have parameter order 
change, state
and this one has
state, change 

which is inconsistent

Comment (piekczyk) [reply features/manageGuniVault/manageGuniVault.ts:139]: I see it now, definitely inconsistency, I will fix it.

**Review by adamskrodzki — APPROVED** [review]

**Pull request merged**